src/schema/v2/article/index.ts

```diff
@@ -53,26 +56,91 @@ export const ArticleType = new GraphQLObjectType<any, ResolverContext>({
       type: new GraphQLList(AuthorType),
       resolve: ({ contributing_authors }) => contributing_authors,
     },
+    description: {
+      type: GraphQLString,
+    },
+    hero: {
+      type: ArticleHero,
+      resolve: ({ hero_section }) => hero_section,
+    },
     href: {
       type: GraphQLString,
       resolve: ({ slug }) => `/article/${slug}`,
     },
+    layout: {
+      type: new GraphQLNonNull(
+        new GraphQLEnumType({
+          name: "ArticleLayout",
+          values: {
+            STANDARD: { value: "standard" },
+            FEATURE: { value: "feature" },
+          },
+        })
+      ),
+    },
+    keywords: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(GraphQLString))
+      ),
+      resolve: ({ keywords }) => (keywords ? keywords : []),
+    },
+    postscript: { type: GraphQLString },
     publishedAt: date,
+    relatedArticles: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(ArticleType))
+      ),
+      resolve: async (
+        { related_article_ids, id, channel_id, tags },
+        _args,
+        { articlesLoader }
+      ) => {
+        const [
+          { results: articlesFeed },
+          { results: relatedArticles },
+        ] = await Promise.all([
+          articlesLoader({
+            channel_id,
+            featured: true,
+            limit: 3,
+            omit: [id, ...(related_article_ids ?? [])],
+            published: true,
+            sort: "-published_at",
+            tags: tags?.length > 0 ? tags : null,
+          }),
+          related_article_ids?.length > 0
+            ? articlesLoader({
+                has_published_media: true,
+                ids: related_article_ids,
+                limit: 3,
+                published: true,
+              })
+            : Promise.resolve({ results: [] }),
+        ])
+
+        return take([...relatedArticles, ...articlesFeed], 3)
+      },
+    },
     sections: {
-      type: new GraphQLList(
-        new GraphQLUnionType({
-          name: "ArticleSectionsType",
-          types: [
-            ArticleSectionCallout,
-            ArticleSectionEmbed,
-            ArticleSectionImageCollection,
-            ArticleSectionImageSet,
-            ArticleSectionSocialEmbed,
-            ArticleSectionText,
-            ArticleSectionVideo,
-          ],
-        })
+      type: new GraphQLNonNull(
```

Comment:
For the future: should we basically always be using new GraphQLNonNull unless its obvious not to?

Reply:
For anything we can assume either won't be null or will have a default, yeah. There's a lot in our schema that could be non-nullable. In particular this crops up annoyingly with arrays of things where we can assume there's always going to be an empty array (or we can default to one).

But yeah this is why everything in our codebase is guarded with `?` even when not necessary.